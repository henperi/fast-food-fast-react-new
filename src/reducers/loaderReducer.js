import { TOGGLE_LOADER } from '../actions/actionTypes';

const initialState = {
  isLoading: false,
};

const loaderReducer = (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_LOADER:
      return {
        isLoading: !state.isLoading,
      };

    default:
      return state;
  }
};

export default loaderReducer;
