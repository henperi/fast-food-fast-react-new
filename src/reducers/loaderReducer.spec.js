/* eslint-disable no-undef */
import toggleLoader from '../actions/loaderAction';

// Reducer to be tested
import loaderReducer from './loaderReducer';

describe('loaderReducer', () => {
  it('should return the default state if no action is specified', () => {
    const initialState = {
      isLoading: false,
    };
    const action = { type: '' };
    expect(loaderReducer(initialState, action)).toMatchSnapshot({});
    expect(loaderReducer(initialState, action)).toEqual(initialState);
  });

  it('should update the state and toggle the loader when type is TOGGLE_LOADER', () => {
    const initialState = {
      isLoading: false,
    };
    const action = toggleLoader();
    expect(loaderReducer(initialState, action)).toMatchSnapshot({});
    expect(loaderReducer(initialState, action)).toEqual({ isLoading: true });
  });
});
