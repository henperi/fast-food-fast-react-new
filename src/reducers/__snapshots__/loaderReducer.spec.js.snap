// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loaderReducer should return the default state if no action is specified 1`] = `
Object {
  "isLoading": false,
}
`;

exports[`loaderReducer should update the state and toggle the loader when type is TOGGLE_LOADER 1`] = `
Object {
  "isLoading": true,
}
`;
